src/containers/habit-tracker/analytics/year-view/Heatmap.tsx

```diff
@@ -0,0 +1,269 @@
+  const daysOfWeek = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
+  const weeksInYear = 52;
+
+  // Filter data to include only entries for the selected year
+  const filteredData = data.filter((entry) => {
+    const entryYear = new Date(entry.date).getFullYear();
+    return entryYear === year;
+  });
+
+  // Get data for weeks and days of week for axis
+  const series = daysOfWeek.map((day, dayIndex) => ({
+    name: day,
+    data: Array.from({ length: weeksInYear }, (_, weekIndex) => {
+      const count =
+        filteredData.find((d) => {
+          const date = new Date(d.date);
+          const weekOfYear = Math.floor(
+            (date.getTime() - new Date(date.getFullYear(), 0, 1).getTime()) /
+              (1000 * 60 * 60 * 24 * 7)
+          );
+          return date.getDay() === dayIndex && weekOfYear === weekIndex;
+        })?.count || 0;
+
+      return { x: weekIndex + 1, y: count };
+    }),
+  }));
+
+  // Options for heatmap chart from Apex Charts
+  const options: ApexOptions = {
+    chart: {
+      height: 350,
+      type: "heatmap",
+      background: colors.bgWhite,
+    },
+    plotOptions: {
+      heatmap: {
+        shadeIntensity: 0.5,
+        radius: 0,
+        useFillColorAsStroke: true,
+        colorScale: {
+          ranges: [
+            {
+              from: 1,
+              to: 1,
+              color: heatMapColors.greenLight,
+              name: "0-1",
+            },
+            {
+              from: 2,
+              to: 3,
+              color: heatMapColors.greenMediumLight,
+              name: "2-3",
+            },
+            {
+              from: 4,
+              to: 5,
+              color: heatMapColors.greenMedium,
+              name: "4-5",
+            },
+            {
+              from: 6,
+              to: 7,
+              color: heatMapColors.greenMediumDark,
+              name: "6-7",
+            },
+            {
+              from: 8,
+              to: 10,
+              color: heatMapColors.greenDark,
+              name: "8-10",
+            },
+            {
+              from: 11,
+              to: 20,
+              color: heatMapColors.greenDarkest,
+              name: "11+",
+            },
+          ],
+        },
+      },
+    },
+    grid: {
+      padding: {
+        top: 10,
+        right: 20,
+        bottom: 10,
+        left: 20,
+      },
+    },
+    dataLabels: {
+      enabled: false,
+    },
+    title: {
+      text: `${habitName} logs for ${year}`,
+    },
+    xaxis: {
+      labels: {
+        formatter: function (value) {
+          const weekNumber = Number(value);
+          const monthLabels: Record<number, string> = {
+            1: "Jan",
+            5: "Feb",
+            9: "Mar",
+            14: "Apr",
+            18: "May",
+            22: "Jun",
+            27: "Jul",
+            31: "Aug",
+            35: "Sep",
+            40: "Oct",
+            44: "Nov",
+            48: "Dec",
+          };
```

Comment:
I am not sure what is all of this logic. Again, seems complicated. Why does this object do exactly where you are defining `1: Jan, 5: Feb` and so on?

Write all the logics yourself. It is a great way to get comfortable with different data structures. Get help from ChatGPT, but don't get the answer.